localstack/services/awslambda/lambda_utils.py

```diff
@@ -288,3 +289,110 @@ def generate_lambda_arn(
+                continue
+            if numeric_filter[idx] == "<" and record_value < float(numeric_filter[idx + 1]):
+                all_comparisons.append(True)
+                continue
+            if numeric_filter[idx] == "<=" and record_value <= float(numeric_filter[idx + 1]):
+                all_comparisons.append(True)
+                continue
+            all_comparisons.append(False)
+        except ValueError:
+            LOG.warn(
+                f"Could not convert filter value {numeric_filter[idx + 1]} to a valid number value for filtering"
+            )
+    return all(all_comparisons)
+
+
+def verify_dict_filter(record_value: any, dict_filter: Dict[str, any]):
+    # https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html#filtering-syntax
+    filter_results = []
+    for key, filter_value in dict_filter.items():
+        if key.lower() == "anything-but":
+            filter_results.append(record_value not in filter_value)
+        elif key.lower() == "numeric":
+            filter_results.append(parse_and_apply_numeric_filter(record_value, filter_value))
+        elif key.lower() == "exists":
+            filter_results.append(
+                bool(filter_value)
+            )  # exists means that the key exists in the event record
+        elif key.lower() == "prefix":
+            filter_results.append(str(record_value.get("S")).startswith(str(filter_value)))
+        else:
+            filter_results.append(False)
+    return all(filter_results)
+
+
+def filter_stream_record(filter_rule: Dict[str, any], record: Dict[str, any]):
+    if not filter_rule:
+        return True
+    # https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html#filtering-syntax
+    filter_results = []
+    for key, value in filter_rule.items():
+        # check if rule exists in event
+        record_value = record.get(key.lower(), record.get(key))
+        if record_value is None:
+            # special case 'exists'
+            if isinstance(value, list) and len(value) > 0:
+                if not value[0].get("exists", True):
+                    filter_results.append(True)
+                    continue
+            filter_results.append(False)
+            continue
+
+        # check if filter rule value is a list (leaf of rule tree) or a dict (rescursively call function)
+        if isinstance(value, list):
+            if len(value) > 0:
+                if isinstance(value[0], (str, int)):
+                    filter_results.append(record_value in value)
+                    continue
+                if isinstance(value[0], dict):
+                    filter_results.append(verify_dict_filter(record_value, value[0]))
+                    continue
+            else:
+                LOG.warn(f"Empty lambda filter: {key}")
+        if isinstance(value, dict):
+            filter_results.append(filter_stream_record(value, record_value))
+            continue
```

Comment:
please avoid the overuse of continue and/or return statements in favor or restructuring the flow